pkg/cloud/services/compute/instance.go

```diff
@@ -505,6 +515,25 @@ func (s *Service) getOrCreatePort(eventObject runtime.Object, clusterName string
 	}
 
 	record.Eventf(eventObject, "SuccessfulCreatePort", "Created port %s with id %s", port.Name, port.ID)
+	if portOpts.DisablePortSecurity == true || openStackCluster.Spec.DisablePortSecurity == true {
+		var portWithPortSecurityExtensions struct {
+			ports.Port
+			portsecurity.PortSecurityExt
+		}
+		err = ports.Get(s.networkClient, port.ID).ExtractInto(&portWithPortSecurityExtensions)
+		if portWithPortSecurityExtensions.PortSecurityEnabled == true {
+			iFalse := false
+			portUpdateOpts := ports.UpdateOpts{}
+			updateOpts := portsecurity.PortUpdateOptsExt{
+				UpdateOptsBuilder:   portUpdateOpts,
+				PortSecurityEnabled: &iFalse,
+			}
+			err := ports.Update(s.networkClient, port.ID, updateOpts).ExtractInto(&portWithPortSecurityExtensions)
```

Comment:
Is there a reason why PortSecurityEnabled can't be set when the port is created? Instead, you first create the port, then get the port and then update it with the PortSecurityEnabled setting. Seems like this may not be necessary, but I'm not sure?

Reply:
I was trying to do that at the beginning, but the field is not available until after the port is created. I.e, the struct we use the create the port do not contain such field. Which is the reason for retrieving the port one more time.
`err = ports.Get(s.networkClient, port.ID).ExtractInto(&portWithPortSecurityExtensions)`

If there was away to add `portWithPortSecurityExtensions` before creation, then it would solve the problem

Reply:
Great, I think then just a comment explaining that this setting must be updated on the port after creation would be best.